### DISCUSSION TITLE
Creating custom getSmoothStepPath

### DISCUSSION THREAD

**Discussion by jonassarcilla** [xyflow/xyflow#2468] [Q&A]
Hi all,

I have a question regarding creating custom edge, it seems not displaying on my side. 
![image](https://user-images.githubusercontent.com/32216903/193774067-f04fb1f8-0e6c-4a9d-baab-3045a3980d08.png)

I use the ff snippet:

const [edgePath, labelX, labelY] = getSmoothStepPath({
    sourceX,
    sourceY,
    sourcePosition,
    targetX,
    targetY,
    targetPosition,
    centerX,
    centerY,
  });

and upon checking the 'edgePath' result, it displays only 'M'.
![image](https://user-images.githubusercontent.com/32216903/193773302-c28312f2-08f5-44bb-a5a8-6935d35aafda.png)

Is there anything I need to add in the params or miss something or wrong in my implementation?

Thanks in advance

Jonas Sarcilla

**Comment by moklick** [comment]
Can you create a codesandbox for this? What are your `centerX` and `centerY` values look like? Could you remove them from the params you pass to `getSmoothStepPath`?

**Reply by jonassarcilla** [reply]
Hi,

upon checking centerX and centerY is undefined
![image](https://user-images.githubusercontent.com/32216903/193791975-b68d9f41-1159-4bd9-bc7e-7ddf37daceb6.png)

The value of edgePath is still 'M' after I remove centerX and centerY from the params
![image](https://user-images.githubusercontent.com/32216903/193792427-95dcefe3-96a2-445e-b3a6-bab306ef620e.png)

**Reply by moklick** [reply]
Can you create a codesandbox? Are you using the latest version 11.1.0 ?
